edk2toolext/environment/self_describing_environment.py

```diff
@@ -54,11 +54,11 @@ def _gather_env_files(self, ext_strings, base_path):
         matches = {}
         for root, dirs, files in os.walk(base_path, topdown=True):
             # Check to see whether any of these directories should be skipped..
-            for index, dir in enumerate(dirs):
-                dir_path = pathlib.Path(root, dir)
-                if dir_path in self.skipped_dirs or dir_path.name == '.git':
-                    logging.debug(f"Skipping: {dir_path.resolve()}")
-                    del dirs[index]
+            dirs[:] = [d
```

Comment:
this syntax is a little surprising.   dirs[:] is basically telling it to replace all elements in the list with the new list from your assignment.  

why not just dirs =

Reply:
The os.walk() documentation requires the list to be modified in-place - https://docs.python.org/3/library/os.html#os.walk

dirs = would create a new list. dirs[:] calls __setitem__() to modify the list in-place, instead of pointing to a new list object.

Reply:
After searching thru the project there is one more place that does this.  It follows a slightly different pattern of iterating a copy and then deleting from the original.  This has the same result and is also safe.  

https://github.com/tianocore/edk2-pytool-library/blob/5b2dbd7b315743caa626c1a4657c642d491ea8c3/edk2toollib/uefi/edk2/guid_list.py#L45